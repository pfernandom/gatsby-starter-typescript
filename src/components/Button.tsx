import React from 'react';

const Button = (props: any) => (
    <button onClick={props.onClick} {...props}>
        {props.children}
    </button>
);

export default Button;
