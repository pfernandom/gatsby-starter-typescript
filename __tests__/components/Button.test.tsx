import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';

import Button from '../../src/components/Button';

describe('Integration | Button', () => {
  it('renders correctly (snapshot)', () => {
    const rendered = renderer.create(
      <Button className="foo">Bar</Button>,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(shallow(<Button className="foo">Bar</Button>).contains(<button className="btn foo">Bar</button>)).toBe(true);
  });
  /*
  it('should be selectable by class "foo"', function() {
    expect(shallow(<Button />).is('.foo')).toBe(true);
  });

  it('should mount in a full DOM', function() {
    expect(mount(<Button />).find('.foo').length).toBe(1);
  });

  it('should render to static HTML', function() {
    expect(render(<Button />).text()).toEqual('Bar');
  });
  */
});
