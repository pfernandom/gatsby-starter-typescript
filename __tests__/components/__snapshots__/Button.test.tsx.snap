// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration | Button renders correctly (snapshot) 1`] = `
<button
  className="btn foo"
>
  Bar
</button>
`;
